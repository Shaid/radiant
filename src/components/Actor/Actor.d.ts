export interface IState {
  actor: any
}

export interface IProps {
}

export interface IDefaultProps {
}

export type PropsWithDefaults = IProps & IDefaultProps
