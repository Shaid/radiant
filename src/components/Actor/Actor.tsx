import React from 'react'
import styled from 'theme'

import Action from 'components/Action'
import { IProps, IDefaultProps, IState } from './Actor.d'

const Actor = styled.div`
  flex: 1 1 20px;
`

const actors = require('data/actors/actors.json')

const getRandomActor = () => actors[Math.floor(Math.random() * actors.length)]

const getActorDescription = (actor: any) => {
  const age = ((a) => {
    if (a < 35) {
      return 'young'
    }

    if (a > 65) {
      return 'old'
    }

    return 'middle-aged'
  })(actor.age)

  const gender = actor.gender === 'female' ? 'woman' : 'man'

  return `A ${age} ${gender}`
}


export default class extends React.PureComponent<IProps, IState> {
  static defaultProps: Partial<IDefaultProps> = {}

  constructor(props: IProps) {
    super(props)

    this.state = {
      actor: getRandomActor()
    }
  }

  interact() {
    console.log(`interacting with actor: ${this.state.actor.name}`)
  }

  render() {
    // const {} = this.props as PropsWithDefaults
    return (
      <Actor>
        <Action onClick={() => this.interact()}>
          {getActorDescription(this.state.actor)} is here.
        </Action>
      </Actor>
    )
  }
}
