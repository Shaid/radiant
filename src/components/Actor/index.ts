export { default } from './Actor'

// NOTE: uncomment this line to pass through named exports
// export * from './Actor'
