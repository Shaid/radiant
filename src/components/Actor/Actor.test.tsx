import React from 'react'
import { shallow } from 'enzyme'

import Actor from '.'

const actor = {
  id: 1,
  name: 'Katarina',
  age: 32,
  gender: 'female',
  appearance: {
    eyes: 'green',
    hair: {
      colour: 'brown',
      length: 'shoulder-length'
    }
  },
  status: 'bored',
  idleActions: [
    'coughs',
    'rolls her eyes',
    'sighs'
  ]
}

describe('<Actor />', () => {
  it('renders', () => {
    const wrapper = shallow(<Actor actor={actor} />)
    expect(wrapper).toBeDefined()
  })

  // @TODO: tests for Actor
})
