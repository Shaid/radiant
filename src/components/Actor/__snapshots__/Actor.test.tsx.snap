// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actor /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <_default
        onClick={[Function]}
      >
        A middle-aged woman
         is here.
      </_default>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "A middle-aged woman",
          " is here.",
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "A middle-aged woman",
        " is here.",
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <_default
          onClick={[Function]}
        >
          A middle-aged woman
           is here.
        </_default>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "A middle-aged woman",
            " is here.",
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "A middle-aged woman",
          " is here.",
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
